--- crfsuite/lib/crf/src/crf1d_context.c.new	2015-04-23 02:42:51.708057503 -0400
+++ crfsuite/lib/crf/src/crf1d_context.c	2015-04-23 02:38:16.676050966 -0400
@@ -514,6 +514,9 @@
     /* Find the node (#T, #i) that reaches EOS with the maximum score. */
     max_score = -FLOAT_MAX;
     prev = ALPHA_SCORE(ctx, T-1);
+    /* Set a score for T-1 to be overwritten later. Just in case we don't
+       end up with something beating -FLOAT_MAX. */
+    labels[T-1] = 0;
     for (i = 0;i < L;++i) {
         if (max_score < prev[i]) {
             max_score = prev[i];
